conmon/conmon.c

```diff
@@ -93,28 +93,39 @@ static GOptionEntry entries[] =
   { NULL }
 };
 
-/* strlen("1997-03-25T13:20:42+01:00") + 1 */
-#define TSBUFLEN 26
+/* strlen("1997-03-25T13:20:42.999999999+01:00") + 1 */
+#define TSBUFLEN 36
 
 int set_k8s_timestamp(char *buf, ssize_t buflen)
 {
 	time_t now = time(NULL);
 	struct tm *tm;
+	struct timespec ts;
 	char off_sign = '+';
 	int off, len, err = -1;
+	int rc;
 
 	if ((tm = localtime(&now)) == NULL)
 		return err;
 
+	if ((rc = clock_gettime(CLOCK_REALTIME, &ts)) < 0) {
```

Comment:
if(clock_gettime(...)) :-) but who's counting chars the optimizer will figure it out.